packages/grid/x-data-grid-pro/src/hooks/features/rowReorder/useGridRowReorder.tsx

```diff
@@ -91,6 +91,10 @@ export const useGridRowReorder = (
         return;
       }
 
+      if (apiRef.current.unstable_isRowPinned(params.id)) {
```

Comment:
I guess it doesn't make much sense to reorder only between the pinned rows (or at least for now)? In the future we can think of an interaction where a user starts dragging a row and that row is dropped into the pinned rows area then it becomes pinned automatically. Same thing can apply for the column pinning.
CC @m4theushw

Reply:
Yeah, I decided to take the same approach as with column pinning. But there's definitely room for improvement for both row pinning and column pinning

Reply:
In the column pinning, I disabled column reordering mostly because it would be pain to support dragging a pinned column out from the pinned columns zone, meaning that it should be unpinned. We can disable row reordering here and wait for someone to ask for it.